ios/PurchasesHybridCommon/PurchasesHybridCommon/SubscriptionInfo+HybridAdditions.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  SubscriptionInfo+HybridAdditions.swift
+//  PurchasesHybridCommon
+//
+//  Created by Toni Rico on 3/3/25.
+//  Copyright © 2025 RevenueCat. All rights reserved.
+//
+
+import Foundation
+import RevenueCat
+
+internal extension SubscriptionInfo {
+
+    var dictionary: [String: Any] {
+        var priceObject: [String: Any]? = nil
+        if let priceCurrency = price?.currency, let priceAmount = price?.amount {
+            priceObject = ["currency": priceCurrency, "amount": priceAmount]
+        }
+        return [
+            "productIdentifier": productIdentifier,
+            "purchaseDate": purchaseDate.rc_formattedAsISO8601(),
+            "originalPurchaseDate": originalPurchaseDate?.rc_formattedAsISO8601() ?? NSNull(),
```

Comment:
I know this is already the same for other objects (e.g. `CustomerInfo`), but I still want to ask. What's the benefit of having
```
key: optionalValue ?? NSNull()
```
instead of
```
key: optionalValue as Any
```
?
To me it seems simpler to just not have the key present if the value is `nil`.

Reply:
Well I can see that this could be a change in the typescript hybrids (RN + Capacitor). As in, the presence of a key but with a null value is different than the abscence of that key 😅. In this case, I think it helps discoverability to have the keys there with null values vs not having the keys at all. This would be my only point in favor of the current system.

As for the other hybrids, I think it doesn't really matter what we do here, as long as we map to the corresponding object correctly...

Reply:
Makes sense, yes. Thanks for clarifying!